src/components/PupilComponents/QuizResultQuestionStem/QuizResultQuestionStem.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  OakCloudinaryImage,
+  OakFlex,
+  OakSpan,
+} from "@oaknational/oak-components";
+
+import {
+  removeMarkdown,
+  shortAnswerTitleFormatter,
+} from "@/components/TeacherComponents/LessonOverviewQuizContainer/quizUtils";
+import {
+  ImageItem,
+  TextItem,
+} from "@/node-lib/curriculum-api-2023/queries/pupilLesson/pupilLesson.schema";
+import { getSizes } from "@/components/SharedComponents/CMSImage/getSizes";
+
+export type QuizQuestionStemProps = {
+  questionStem: (ImageItem | TextItem)[];
+  index: number;
+  takeFullHeight?: boolean;
+};
+
+export const QuizResultQuestionStem = (props: QuizQuestionStemProps) => {
+  const { questionStem, index, takeFullHeight } = props;
+  const displayNumber = `Q${index + 1}.`;
+
+  return (
+    <OakFlex
+      $flexDirection={"column"}
+      $gap={"space-between-m"}
+      $color={"text-primary"}
+      $height={takeFullHeight ? "100%" : "auto"}
+      $justifyContent={["center", "flex-start"]}
+    >
+      <OakFlex
+        key="stem-header"
+        $mt={
+          takeFullHeight
+            ? ["space-between-none", "space-between-xl", "space-between-xxl"]
+            : "space-between-none"
+        }
+      >
+        {questionStem[0]?.type === "text" && (
+          <OakSpan
+            key={`q-${displayNumber}-stem-element-0`}
+            $font={"body-2-bold"}
+          >
+            {`${displayNumber} ${shortAnswerTitleFormatter(
+              removeMarkdown(questionStem[0].text),
+            )}`}
+          </OakSpan>
+        )}
+      </OakFlex>
+
+      {questionStem.map((stemItem, i) => {
```

Comment:
If SonarCloud is still complaining then this is probably the portion that could be refactored to a sub component used by both.

Reply:
its not

Reply:
I meant after you've made my suggested change